meteor/client/ui/Settings/ConfigManifestSettings.tsx

```diff
@@ -84,14 +86,82 @@ function filterLayerMappings(
 	return result
 }
 
+function getFilteredTableValuesFromComparison<DBInterface extends { _id: ProtectedString<any> }>(
+	targetFullAttribute: string,
+	item: ConfigManifestEntrySelectFromTableEntryWithComparisonMappings<boolean>,
+	configPath: string,
+	dbObject: DBInterface,
+	alternateDbObject?: DBInterface
+): SelectOption[] {
+	const sourceAttribute = `${configPath}.${item.sourceTableId}`
+	const targetAttribute = `${targetFullAttribute}.${0}`
+
+	const sourceTable: TableConfigItemValue[] =
+		objectPathGet(dbObject, sourceAttribute) ?? objectPathGet(alternateDbObject, sourceAttribute)
+	const targetRow: TableConfigItemValue =
+		objectPathGet(dbObject, targetAttribute) ?? objectPathGet(alternateDbObject, targetAttribute)
+
+	if (!Array.isArray(sourceTable) || !targetRow) {
+		return []
+	}
+
+	const result = sourceTable.flatMap((sourceRow) => {
+		if (!hasRowEntryWithValue(sourceRow, item.sourceColumnIdWithValue)) {
+			return []
+		}
+		const comparisonMapping = getComparisonMapping(item.comparisonMappings, sourceRow)
+		if (!comparisonMapping) {
+			return []
+		}
+
+		return getSourceValuesFromComparisonMapping(sourceRow, targetRow, item.sourceColumnIdWithValue, comparisonMapping)
+	})
```

Comment:
To reduce the complexity of this function, the lambda can be extracted, such that this becomes:
```typescript
	const options = sourceTable.flatMap((sourceRow) => getOptionsFromSourceRow(sourceRow, targetRow, item))
```

Reply:
Updated!